app/models/git_repository.rb

```diff
@@ -45,7 +45,12 @@ def tag_from_ref(git_reference)
         io.read.strip
       end
 
-      tag if $?.success?
+      if $?.success?
+        tag
+      else
+        # no tags present
+        commit_from_ref(git_reference)
+      end
```

Comment:
If there's no tag associated with a commit, I would assume `tag_from_ref` would return nil instead of having a fallback.

I think I would leave this as-is, and update `JobExecution#setup` to look like:

```ruby
tag = @repository.tag_from_ref(@reference) || @repository.commit_from_ref(@reference)
```